docs/docs/more-info/architecture.mdx

```diff
@@ -0,0 +1,83 @@
+
+* Packages may be executed in a cross platform way (on Window, OSX or Linux)
+* Complexity should be hidden from the user to make spinning up packages easy
+* The software that executes the packages should be self contained so that it is easy to download and get running
+
+The core concept of the execution architecture is that all the tooling to execute packages in contained within a single docker image - called the execution image. This enables it to run cross platform as long as the required dependencies are installed (Docker Desktop for Windows/OSX or Docker engine and Docker compose for Linux). To enable the execution image to spin up packages using docker it is passed the docker socket file from the host so that any containers it created are in fact created on the host. It may also be passed other configuration form the host for other platform such as Kubernetes to allow it to deploy packages to those platform as well using the users current cluster configuration. The execution image give us a consistent environment to develop for and allow us to run the scripts that execute the packages without caring about the underlying host OS, we just need to interface with the deployment platform.
+
+An overview of the architecture is displayed below. Following that we describe how these components all works together in the sections that follow.
+
+<div class="text--center">
+  <img alt="Package architecture" src={useBaseUrl('img/instant-openhie-arch.png')} />
+</div>
+
+## Go executable CLI app
+
+The CLI app will allow the setup and configuration of Instant OpenHIE to be easily managed by the user. It is the only executable that the user would need to download to interact with instant OpenHIE. It is written in Go so that it may compile to a platform-independent executable with 0 dependencies. It will know how to fetch the execution image from docker hub and be able to execute it at the command of the user.
+
+The app will allow the user to execute it as an interactive CLI tool, however, it will also have the ability to launch a Web UI and API server that work together to provide the user with a Web interface for managing and executing Instant OpenHIE. Choices made in the Web UI will be sent to the API server and from there the server will issue commands by running the execution image with a particular set of parameters as shown in the diagram.
+
+The CLI app will also be responsible for ensuring the execution image is executed with the correct configuration to ensure it works cross-platform and with the necessary user setting. For example: it will mount the docker socket file into the container so that docker container are created on the host and it will mount various user config files for Kubernetes, docker hub and/or AWS credentials. This enables the execution image to execute as if it were configured like the users host system.
```

Comment:
```suggestion
The CLI app will also be responsible for ensuring the execution image is executed with the correct configuration to ensure it works cross-platform and with the necessary user setting. For example, it will mount the docker socket file into the container so that docker containers are created on the host and it will mount various user config files for Kubernetes, docker hub and/or AWS credentials. This enables the execution image to execute as if it were configured like the user's host system.
```